src/app/tasks/shared/tasks.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Task, TaskApi } from './task.model';
+import { TASKS } from './mock-tasks';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TasksService {
+  getTasks(
+    sortKey?: keyof Task,
+    order?: string,
+    pageItems?: number,
+    pageNumber?: number
+  ): Promise<TaskApi> {
+    let results: Task[] = [...TASKS];
+
+    if (sortKey && order) {
+      results = results.sort((a: Task, b: Task) => {
+        if (a[sortKey] > b[sortKey]) {
+          return order === 'asc' ? 1 : -1;
+        }
+        if (b[sortKey] > a[sortKey]) {
```

Comment:
'asc' could be added to a constant, and use the constant everywhere is needed.

Reply:
updated